manual_benchmarks/test_datatypes_crunch.ipynb

```diff
@@ -0,0 +1,283 @@
+        "            print(\"******************************\")\n",
+        "            print(\"Column: \",col)\n",
+        "            print(\"dtype before: \",props[col].dtype)\n",
+        "            \n",
+        "            # make variables for Int, max and min\n",
+        "            IsInt = False\n",
+        "            mx = props[col].max()\n",
+        "            mn = props[col].min()\n",
+        "            \n",
+        "            # Integer does not support NA, therefore, NA needs to be filled\n",
+        "            if not np.isfinite(props[col]).all(): \n",
+        "                NAlist.append(col)\n",
+        "                props[col].fillna(-1,inplace=True)  \n",
+        "                   \n",
+        "            # test if column can be converted to an integer\n",
+        "            asint = props[col].fillna(0).astype(np.int64)\n",
+        "            result = (props[col] - asint)\n",
+        "            result = result.sum()\n",
+        "            if result > -0.01 and result < 0.01:\n",
+        "                IsInt = True\n",
+        "\n",
+        "            \n",
+        "            # Make Integer/unsigned Integer datatypes\n",
+        "            if IsInt:\n",
+        "                if mn >= 0:\n",
+        "                    if mx < 255:\n",
+        "                        props[col] = props[col].astype(np.uint8)\n",
+        "                    elif mx < 65535:\n",
+        "                        props[col] = props[col].astype(np.uint16)\n",
+        "                    elif mx < 4294967295:\n",
+        "                        props[col] = props[col].astype(np.uint32)\n",
+        "                    else:\n",
+        "                        props[col] = props[col].astype(np.uint64)\n",
+        "                else:\n",
+        "                    if mn > np.iinfo(np.int8).min and mx < np.iinfo(np.int8).max:\n",
+        "                        props[col] = props[col].astype(np.int8)\n",
+        "                    elif mn > np.iinfo(np.int16).min and mx < np.iinfo(np.int16).max:\n",
+        "                        props[col] = props[col].astype(np.int16)\n",
+        "                    elif mn > np.iinfo(np.int32).min and mx < np.iinfo(np.int32).max:\n",
+        "                        props[col] = props[col].astype(np.int32)\n",
+        "                    elif mn > np.iinfo(np.int64).min and mx < np.iinfo(np.int64).max:\n",
+        "                        props[col] = props[col].astype(np.int64)    \n",
+        "            \n",
+        "            # Make float datatypes 32 bit\n",
+        "            else:\n",
+        "                props[col] = props[col].astype(np.float32)\n",
+        "            \n",
+        "            # Print new column type\n",
+        "            print(\"dtype after: \",props[col].dtype)\n",
+        "            print(\"******************************\")\n",
+        "    \n",
```

Comment:
As @Hajime Kawahara recently pointed out, conversion to float32 may be too restrictive for the line position (v0) https://radis-radiation.slack.com/archives/C01G3117DJS/p1625716717008500?thread_ts=1625633254.001800&cid=C01G3117DJS. For all other decimals float32 is probably fine.

Reply:
BTW,  cool function :  np.finfo . Shows the relative accuracy, it may be used to check if our dtype reduction are valid? (as we know the number of significant digits used in the HITRAN database) 
https://stackoverflow.com/questions/39134956/accuracy-of-float32